### ISSUE TITLE
False negative: missing numpy param doc when "default ..." in same line of "param : type"

### ISSUE THREAD

**Issue by vepain** [PyCQA/pylint#6211]
### Bug description

False negative : 
Warning "missing param doc" should'nt be raised when these type of docstring format:
```python
def func(x: int = 0):
  """This is func.

  Parameter
  ---------
  x : int, default 0  # this raises the warning
    The x parameter
  """
  # ...
```

### Configuration

```ini
[MASTER]
load-plugins =
    pylint.extensions.docstyle,
    pylint.extensions.docparams

[MESSAGES CONTROL]
enable = W

[BASIC]
accept-no-param-doc = no
```

### Command used

```shell
pylint --rcfile=tox.ini a.py
```

### Pylint output

```shell
"orientation" missing in parameter documentation
```

### Expected behavior

No warning

### Pylint version

```shell
pylint 2.13.5
astroid 2.11.2
Python 3.9.12 (main, Mar 24 2022, 16:20:11) 
[GCC 9.4.0]
```

### OS / Environment

Ubuntu 20.04

**Comment by lkev** [comment]
+1 having the same issue.

Just to add - the following does not raise any error, so maybe just allow items in parenthesis and/or the "default" word after the comma?

```python
def func(x: int = 0):
  """This is func.

  Parameter
  ---------
  x : int, optional
    The x parameter, by default 0
  """
  # ...
```

**Comment by DanielNoord** [comment]
See https://numpydoc.readthedocs.io/en/latest/format.html#parameters and the failures in https://github.com/PyCQA/pylint/pull/7149.

**Comment by adam-grant-hendry** [comment]
I am seeing this as well with

```
python 3.8.10
pylint 2.14.5
```

using

```pylint.extensions.docparams```

@DanielNoord Has this not been fixed yet?

**Comment by DanielNoord** [comment]
No, I don't think so. The linked PR fixed something different.

**Comment by adam-grant-hendry** [comment]
@DanielNoord Can you point me to the section of code to review? I can submit a PR if you like.

**Comment by adam-grant-hendry** [comment]
> @DanielNoord Can you point me to the section of code to review? I can submit a PR if you like.

Actually, I'm assuming this is [pylint/extensions/docparams.py](https://github.com/PyCQA/pylint/blob/main/pylint/extensions/docparams.py). I'll submit a PR.

**Comment by Pierre-Sassoulas** [comment]
That's right @adam-grant-hendry, most tests for this checker are in https://github.com/PyCQA/pylint/tree/main/tests/functional/ext/docparams

**Comment by adam-grant-hendry** [comment]
@lkev @vepain One thing to note: you are using `Parameter` instead of `Parameters` (plural). Change this to `Parameters` and see if that helps because `pylint` checks for sections using the plural form:

```python
re_param_section = re.compile(
        _re_section_template.format(r"(?:Args|Arguments|Parameters)"),
        re.X | re.S | re.M,
    )
```

and this is how it is written in the [`numpydoc` Style Guide](https://numpydoc.readthedocs.io/en/latest/format.html#parameters).
